### PR TITLE
Add cpp base image

### PR DESCRIPTION
**Pull request by doosuu**
Adds a base devContainer image for C++

### CODE REVIEW

File: Dockerfiles/Cpp/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+COPY ./common/ /scripts/
+
+WORKDIR /scripts
+RUN ./install-python.sh && \
+    ./install-dapr-cli.sh && \
+    ./install-k3d-prerequisites.sh \
+    ./install-k3d-tooling.sh
+
+RUN pip3 install gcovr conan pre-commit cpplint
```
Comment (MP91) [inline Dockerfiles/Cpp/Dockerfile]: we should use the requirements.txt here and also set the versions to be fixed

**Review by MP91 — CHANGES_REQUESTED** [review]

File: Dockerfiles/Cpp/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+COPY ./common/ /scripts/
+
+WORKDIR /scripts
+RUN ./install-python.sh && \
+    ./install-dapr-cli.sh && \
+    ./install-k3d-prerequisites.sh \
+    ./install-k3d-tooling.sh
+
+ADD --chown=4000 https://raw.githubusercontent.com/eclipse-velocitas/vehicle-app-cpp-sdk/main/requirements.txt  /tmp/
+USER 4000
```
Comment (MP91) [inline Dockerfiles/Cpp/Dockerfile:100]: We need to set the USER back to root at the end, otherwise we get errors when using this in our repo.

**Review by MP91 — CHANGES_REQUESTED** [review]

File: Dockerfiles/Cpp/Dockerfile (hunk 1 above)
Comment (doosuu) [reply Dockerfiles/Cpp/Dockerfile]: Versions will be fixed in next SDK PR, but requirements.txt is now sourced.

File: Dockerfiles/Cpp/Dockerfile (hunk 2 above)
Comment (doosuu) [reply Dockerfiles/Cpp/Dockerfile:100]: Fixed.

**Review by MP91 — APPROVED** [review]
LGTM

**Pull request merged**